Huge Chambray Perch

medium

# External calls like .transfer functions should have checks

## Summary
In Solidity, when making an external call to transfer Ether, such as using the .call/.transfer method, it is crucial to check the return value of the call to ensure that the transfer was successful. 

## Vulnerability Detail
In the MockweEth.sol 

     
    function wrap(uint256 _eETHAmount) external returns (uint256) {
        eETH.transferFrom(msg.sender, address(this), _eETHAmount);
    }


    function unwrap(uint256 _weETHAmount) external returns (uint256) {
        eETH.transfer(msg.sender, returnAmount);   
    }

It is crucial to check the return value of the call to ensure that the transfer was successful. Failure to do so can result in undetected failures of Ether transfers, potentially leading to loss of funds or incorrect contract state.Leading to false transactions and an attacker could transfer false funds.
## Impact
As mentioned above in the details section . the impact could be severe as if the fund transfers fails from `msg.sender` account the function would return the calculated amount but the actual amount is not transfered.  
## Code Snippet
`MockSDAI.deposit(uint256,address) (contracts/mocks/MockSDAI.sol#18-28)`
`MockSDAI.redeem(uint256,address,address) (contracts/mocks/MockSDAI.sol#30-50)` 
`MockWstETH.unwrap(uint256) (contracts/mocks/MockWstETH.sol#30-40) `
`MockWeETH.wrap(uint256) (contracts/mocks/MockweETH.sol#18-28)` 
`MockWeETH.unwrap(uint256) (contracts/mocks/MockweETH.sol#30-40)` 
## Tool used

Manual Review,slither

## Recommendation
we should account for the output value from transfer function and then change the state.

    (bool success,)= eETH.transfer(msg.sender, returnAmount);
    require(success,"transfer failed")